baseband/vdif/header.py

```diff
@@ -28,6 +32,41 @@
                   precision=9)
 
 
+_vdif_edv_header_classes = {}
+"""Dict for storing pairs of VDIF EDV and header classes."""
+
+class _VDIFHeaderRegistry(type):
+    """
+    Registry of VDIF Header EDV types, using the ``_vdif_edv_header_classes``
+    dict.  Checks for keyword and subclass conflicts before registering.
+    """
+    _vdif_edv_header_classes = _vdif_edv_header_classes
```

Comment:
Just to follow `astropy.time.TimeFormatMeta` more closely, I vote to keep this in, changing the variable's name to `_registry`.